### ISSUE TITLE
[Rule change] allow to use block body and lambdas in avoid-returning-widgets 

### ISSUE THREAD

**Issue by grafovdenis** [dart-code-checker/dart-code-metrics#401]
**What rule do you want to change?**

[avoid-returning-widgets](https://github.com/dart-code-checker/dart-code-metrics/blob/master/doc/rules/avoid-returning-widgets.md)

**How will the change be implemented? (New option, new default behavior, etc.)?**

Allow to use block body and lambdas in builder.

Currently, only tear-off closure is accepted to pass avoid-returning-widgets rule.

**Please provide some example code that this change will affect:**

```dart
class MyWidget extends StatelessWidget {
  Widget _buildMyWidget(BuildContext context) {
    return Container();
  }

  @override
  Widget build(BuildContext context) {
    return Builder(
      builder: (context) => _buildMyWidget(context),
    );
  }
}
```

```dart
class MyWidget extends StatelessWidget {
  Widget _buildMyWidget(BuildContext context) {
    return Container();
  }

  @override
  Widget build(BuildContext context) {
    return Builder(
      builder: (context) { 
        return _buildMyWidget(context);
      },
    );
  }
}
```

**What does the rule currently do for this code?**

Lints `builder: (context) => _buildMyWidget(context),` and `return _buildMyWidget(context);`

**Comment by dkrutskikh** [comment]
Available in 4.11.0 release 🚀
